framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism.spec.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const BlockSynchronizationMechanism = require('../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+
+describe('BlockSynchronizationMechanism', () => {
+	const stubs = {
+		channel: {
+			invoke: jest.fn(),
+		},
+		modules: {
+			blocks: {
+				lastBlock: jest.fn(),
+			},
+		},
+	};
+	let blockSynchronizationMechanism;
+
+	beforeEach(() => {
+		blockSynchronizationMechanism = new BlockSynchronizationMechanism({
+			channel: stubs.channel,
+			modules: stubs.modules,
+		});
+	});
+
+	describe('#constructor', () => {});
+
+	describe('#_computeBestPeer', () => {
+		it('should accept an array of peers as input', () => {
+			expect(blockSynchronizationMechanism._computeBestPeer.length).toEqual(1);
+		});
+
+		it('should return a peer object', () => {
+			stubs.modules.blocks.lastBlock = {
+				height: 0,
+				prevotedConfirmedUptoHeight: 0,
+			};
+
+			const peers = [
+				{
+					lastBlockId: '12343245',
+					prevotedConfirmedUptoHeight: 1,
+					height: 67,
+					ip: '127.0.0.1',
+				},
```

Comment:
I'm just speculating now, maybe putting another peer object into the peers array could improve the test. Because it's returning one of the good peers randomly, right? so, this test doesn't cover that scenario properly. We need to give 2 good and 1 bad peers and make sure that the best peer is always one of those 2 good peers.

Reply:
Good catch. This behavior is being tested at line 85 for a bigger set of peers. so the scenario is covered.